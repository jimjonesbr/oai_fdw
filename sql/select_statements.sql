CREATE FOREIGN TABLE dnb_zdb_oai_dc (
  id text OPTIONS (oai_node 'identifier'),
  content text OPTIONS (oai_node 'content'),
  setspec text[] OPTIONS (oai_node 'setspec'),
  datestamp timestamp OPTIONS (oai_node 'datestamp'),
  meta text OPTIONS (oai_node 'metadataprefix')
 )
SERVER oai_server_dnb OPTIONS (setspec 'zdb', 
                               metadataPrefix 'oai_dc');

--override 'setspec' option
--modify 'metadataPrefix'
SELECT ROW_NUMBER() OVER (), * FROM dnb_zdb_oai_dc
WHERE
  meta = 'MARC21-xml' AND
  datestamp BETWEEN '2022-02-01' AND '2022-02-02' AND
  setspec <@ ARRAY['dnb:reiheC'];


--override 'set' option
--maps January,2021
SELECT ROW_NUMBER() OVER (), * 
FROM dnb_zdb_oai_dc
WHERE
  datestamp BETWEEN '2021-01-01' AND '2021-01-31' AND
  setspec <@ ARRAY['dnb:reiheC'];

-- counting (computed locally)
SELECT count(*) 
FROM dnb_zdb_oai_dc
WHERE
  datestamp BETWEEN '2021-01-01' AND '2021-01-31' AND
  setspec <@ ARRAY['dnb:reiheC'];
  
-- GetRecord request
SELECT * FROM dnb_zdb_oai_dc
WHERE id = 'oai:dnb.de/zdb/1250800153';


-- ListIdentifiers request
SELECT id,datestamp,meta,setspec 
FROM dnb_zdb_oai_dc
WHERE datestamp BETWEEN '2021-01-01' AND '2021-01-03';

-- SELECT using timestamps hh24:mm:ss hh24:mi:ss
SELECT * FROM davidrumsey_oai_dc 
WHERE updatedate BETWEEN '2022-07-04 03:23:24' AND '2022-07-04 03:23:25';
