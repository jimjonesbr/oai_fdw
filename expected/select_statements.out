CREATE FOREIGN TABLE dnb_zdb_oai_dc (
  id text OPTIONS (oai_node 'identifier'),
  content text OPTIONS (oai_node 'content'),
  setspec text[] OPTIONS (oai_node 'setspec'),
  datestamp timestamp OPTIONS (oai_node 'datestamp'),
  meta text OPTIONS (oai_node 'metadataprefix')
 )
SERVER oai_server_dnb OPTIONS (setspec 'zdb', 
                               metadataPrefix 'oai_dc');
--override 'setspec' option
--modify 'metadataPrefix'
CREATE TABLE tb_dnb_zdb_oai_dc AS
SELECT * FROM dnb_zdb_oai_dc
WHERE
  meta = 'MARC21-xml' AND
  datestamp BETWEEN '2022-02-01' AND '2022-02-02' AND
  setspec <@ ARRAY['dnb:reiheC'];
SELECT
  count(*) = count(distinct id) records_are_distinct,
  count(*) > 0 AS not_empty
FROM tb_dnb_zdb_oai_dc;
 records_are_distinct | not_empty 
----------------------+-----------
 t                    | t
(1 row)

--override 'set' option
--maps January,2021
CREATE TABLE tb_dnb_maps_marc21 AS
SELECT * FROM dnb_zdb_oai_dc
WHERE
  datestamp BETWEEN '2021-01-01' AND '2021-01-31' AND
  setspec <@ ARRAY['dnb:reiheC'];
SELECT
  count(*) = count(distinct id) records_are_distinct,
  count(*) > 0 AS not_empty
FROM tb_dnb_maps_marc21;
 records_are_distinct | not_empty 
----------------------+-----------
 t                    | t
(1 row)

-- GetRecord request
SELECT * FROM dnb_zdb_oai_dc
WHERE id = 'oai:dnb.de/zdb/1250800153';
            id             |                                                                      content                                                                       | setspec |        datestamp         |  meta  
---------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------+---------+--------------------------+--------
 oai:dnb.de/zdb/1250800153 | <dc xmlns:dnb="http://d-nb.de/standards/dnbterms" xmlns="http://www.openarchives.org/OAI/2.0/oai_dc/" xmlns:dc="http://purl.org/dc/elements/1.1/">+| {zdb}   | Tue Feb 01 15:39:05 2022 | oai_dc
                           |   <dc:title>Jahrbuch Deutsch als Fremdsprache</dc:title>                                                                                          +|         |                          | 
                           |   <dc:publisher>MÃ¼nchen : Iudicium Verlag</dc:publisher>                                                                                          +|         |                          | 
                           |   <dc:date>2022</dc:date>                                                                                                                         +|         |                          | 
                           |   <dc:language>ger</dc:language>                                                                                                                  +|         |                          | 
                           |   <dc:identifier xsi:type="dnb:IDN">1250800153</dc:identifier>                                                                                    +|         |                          | 
                           |   <dc:identifier xsi:type="dnb:ZDBID">3108310-9</dc:identifier>                                                                                   +|         |                          | 
                           |   <dc:type>Online-Ressource</dc:type>                                                                                                             +|         |                          | 
                           |   <dc:relation>http://d-nb.info/011071060</dc:relation>                                                                                           +|         |                          | 
                           | </dc>                                                                                                                                              |         |                          | 
(1 row)

